### ISSUE TITLE
Refactor EventSequence Grain

### ISSUE THREAD

**Issue by einari** [Cratis/Chronicle#1451]
The EventSequence Grain is doing too much. It is hard to follow the code and it also impacts performance negatively.

It should be split into at least two grains. The management of appending should be put into its own grain that optimizes the appends into batches that it can append to the underlying storage engine.

However, this batching grain needs to be work with transactions.
Keep #1014 and #960 in mind.

The EventSequence grain will be responsible for validating the incoming events, while the batch management and actual appending happen in a separate grain. We should leverage [Orleans local placement](https://learn.microsoft.com/en-us/dotnet/orleans/grains/grain-placement#local-placement) for this to avoid unnecessary latency when working with single events, which will be the majority of cases. 

Also important to keep in mind that we do return a result object with the sequence numbers of the appended events. 

By doing this, we can allow for call interleaving on the append methods. We might not want to have the entire grain be re-entrant, but permit the append methods to be [interleaved](https://learn.microsoft.com/en-us/dotnet/orleans/grains/request-scheduling#interleaving-methods).

**Comment by woksin** [comment]
Closing as part of an org-wide stale-issue sweep: no activity in over 12 months (excluding a 2025-04-11 bulk label pass). If this is still relevant, please reopen or file a fresh issue with current context.

**Comment by woksin** [comment]
Reopened: the automated stale sweep was too aggressive — many of these issues intentionally serve as long-lived reminders. Labeled `stale-review` for human re-triage.
